library/src/main/java/com/novoda/downloadmanager/LiteDownloadManagerDownloader.java

```diff
@@ -69,38 +68,19 @@ public void download(Batch batch, Map<DownloadBatchId, DownloadBatch> downloadBa
 
     public void download(DownloadBatch downloadBatch, Map<DownloadBatchId, DownloadBatch> downloadBatchMap) {
         downloadBatchMap.put(downloadBatch.getId(), downloadBatch);
-        if (downloadService == null) {
-            ensureDownloadServiceExistsAndDownload(downloadBatch);
-        } else {
-            executeDownload(downloadBatch);
-        }
+        executor.submit(() -> WaitForDownloadService.<Void>waitFor(downloadService, waitForDownloadService)
```

Comment:
fine for me, its exactly the same as in the DownloadManager with exactly the same objects because they are shared 👍 

This submission won't block this current executor because it submits into a dedicated download thread on the Service itself. So this submits and ends fast.